app/models/miq_filter.rb

```diff
@@ -44,27 +44,18 @@ def self.belongsto2object_list(tag)
     raise _("invalid tag: %{tag}") % {:tag => tag} unless tag.starts_with?("/belongsto/ExtManagementSystem")
     parts = tag.split("/")[2..-1]
 
-    # root object
-    klass, name = parts.shift.split("|")
-    obj = find_object_by_name(klass, name)
-
-    if obj.nil?
-      _log.warn("lookup for klass=#{klass.to_s.inspect} with name=#{name.inspect} failed in tag=#{tag.inspect}")
+    # Get the full path from the leaf object to the root
+    klass, name = parts.last.split("|", 2)
+    result = find_object_by_name(klass, name)&.path
```

Comment:
how do you deal with 2 same-named objects (cluster, folder, vm) in different EMSes?  It seems you have to find all the objects matching the name and then find the right one via your path approach.

Reply:
That's why I marked it wip (see wip comment on the 2 caveats).  I just wanted to get something out there to play with first.

Reply:
Updated